repo/blob/filesystem/filesystem_storage.go

```diff
@@ -70,6 +70,22 @@ func (fs *fsImpl) isRetriable(err error) bool {
 	return errors.Is(err, errRetriableInvalidLength)
 }
 
+func (fs *fsStorage) GetCapacity(ctx context.Context) (blob.Capacity, error) {
+	c, err := retry.WithExponentialBackoff(ctx, "GetCapacity", func() (interface{}, error) {
+		stat, err := fs.Impl.(*fsImpl).osi.Statfs("/")
```

Comment:
this will return the root filesystem capacity, not necessarily the capacity of the actual blob storage location, is that intended?

Reply:
Thanks for catching this @jkowalski , I don't think I intended it that way–I had a misconception that a repo would have all the frees space on a volume available to it, which is not true, not least because quota policies can implemented as @talios suggested. I'll change these to use the repo path.